rapid7_insightvm/komand_rapid7_insightvm/actions/get_asset_vulnerabilities/action.py

```diff
@@ -17,8 +19,69 @@ def __init__(self):
         resources = resource_helper.paged_resource_request(endpoint=endpoint,
                                                            method='get')
-        return {Output.VULNERABILITIES: resources}
+        if not risk_score:
+            return {Output.VULNERABILITIES: resources}
+        else:
+            resources = self.get_vulnerabilities(resources)
+            return {Output.VULNERABILITIES: resources}
+
+    async def async_get_vulnerabilities(self, vuln_ids: list) -> list:
+        """
+        Prepares a list of vulnerability ids for asynchronous API calls
+        :param vuln_ids: A list of vulnerability ids
+        :return: A list of vulnerability details
+        """
+        connection = self.connection.async_connection
+        async with connection.get_async_session() as async_session:
+            tasks: [asyncio.Future] = []
+            for vuln_id in vuln_ids:
+                endpoint = endpoints.Vulnerability.vulnerability(self.connection.console_url, vuln_id)
+                tasks.append(asyncio.ensure_future(connection.async_request(session=async_session,
+                                                                            endpoint=endpoint, method='get')))
+            vulnerabilities = await asyncio.gather(*tasks)
+            return vulnerabilities
+
+    def get_vulnerabilities(self, resources: list) -> list:
+        """
+        Finds risk scores for a list of asset_vulnerabilities and attaches them
+        :param resources: A list of asset_vulnerabilities
+        :return: A list of asset_vulnerabilities with risk scores
+        """
+        vuln_ids = list()
+        risk_score = list()
+        for resource in resources:
+            try:
+                vuln_ids.append(resource['id'])
+            except KeyError:
+                self.logger.error(f'The following resource did not have an ID:\n{resource}')
+                continue
+
+        vulnerabilities = asyncio.run(self.async_get_vulnerabilities(vuln_ids))
+
+        for vulnerability in vulnerabilities:
+            risk_score.append({'id': vulnerability['id'], 'riskScore': vulnerability['riskScore']})
+
+        sorted_resources = self.add_risk_scores_to_resources(resources, risk_score)
+        return sorted_resources
+
+    @staticmethod
+    def add_risk_scores_to_resources(resources: list, risk_scores: list) -> list:
+        """
+        Sorts asset_vulnerabilities and appends risk scores to them. Returns asset_vulnerabilities list
+        sorted by risk score descending
+        :param resources: A list of asset_vulnerabilities
+        :param risk_scores: A list of risk scores
+        :return: A sorted list of asset_vulnerabilities with risk scores attached
+        """
+        sorted_resources = sorted(resources, key=itemgetter('id'))
+        sorted_risk_score = sorted(risk_scores, key=itemgetter('id'))
+        for i in range(0, len(sorted_resources), 1):
```

Comment:
You don't have to fix this, but for future reference I'd do 

```suggestion
        for i, item in enumerate(sorted_resources):
```

Then you don't have to mess with the classic for loop style.